pkg/workflows/upgrade.go

```diff
@@ -590,20 +599,23 @@ func (s *resumeFluxReconcile) Run(ctx context.Context, commandContext *task.Comm
 		commandContext.SetError(err)
 		return &writeClusterConfigTask{}
 	}
+	if !s.eksaSpecDiff {
```

Comment:
what's the use case of this? if eksa spec is different, we skip writeclusterconfig and delete bootstrap?

Reply:
If there is no difference in the eksaSpec, then we skip writeclusterconfig and delete bootstrap. Since we pause reconciliation before we decide whether or not the spec changed, we have to resume reconciliation if it didn't change & we wouldn't have to write a new cluster config & there would be no new bootstrap cluster to delete